src/main/java/matching_system/MatchManager.java

```diff
@@ -0,0 +1,36 @@
+package matching_system;
+
+import data_access_storage.RequestModel;
+import data_access_storage.UserRepoManager;
+import entities.Matcher;
+import entities.UserAccount;
+
+import java.util.ArrayList;
+
+public class MatchManager implements MatcherInputBoundary{
+    private UserAccount user;
+    private Matcher match;
+
+    final UserRepoManager userRepoManager;
+
+    public MatchManager(MatchRequestModel matchRequestModel, UserRepoManager userRepoManager){
+        this.user = userRepoManager.getUserAccount(matchRequestModel.getUserEmail()).getUserAccount();
+        this.userRepoManager = userRepoManager;
+        ArrayList<UserAccount> userAccountArrayList = new ArrayList<UserAccount>();
+        for(RequestModel rm: this.userRepoManager.getAllUserAccount().values()){
+            userAccountArrayList.add(rm.getUserAccount());
+        }
+        this.match = new Matcher(user,userAccountArrayList);
+    }
+
+    @Override
+    public MatchResponseModel create(MatchRequestModel requestModel){
+        UserAccount[] matchedUsers = match.getMatches();
+        ArrayList<UserData> matchedData = new ArrayList<UserData>();
+        for(UserAccount user: matchedUsers){
+            matchedData.add(new UserData(user));
```

Comment:
In this case, as I am creating UserData using user from matchedUsers, which are all internal, how do I inject them? Will I need to send matchedUsers out, create UserDatas, and then send them back in?

Reply:
> "New UserData(User)" creates a hard dependency from MatchManager to UserData, which we want to avoid. See design pattern dependency injection for solution.

In this case, as I am creating UserData using user from matchedUsers, which are all internal, how do I inject them? Will I need to send matchedUsers out, create UserDatas, and then send them back in?

Reply:
Actually, I looked at your code again. You already have a "add" method from matchedData. That means the add method can be tested.  So this is not a problem. Sorry for the confusion.